lib/modules/coverage.js

```diff
@@ -673,92 +642,125 @@ internals.file = async function (filename, data, options) {
 
         num++;
 
-        let isMiss = false;
-        ret.source[num] = {
-            source: line
+        const source = {
+            source: line,
+            hits: data.lines[num],
+            miss: false
         };
 
+        ret.source[num] = source;
+
+        const isValidLine = !data.commentedLines[num] && line.trim();
+
+        if (smc) {
+            ret.sourcemaps = true;
+            const position = {
+                source: ret.filename,
+                line: num,
+                column: source.source.length    // when using 0 column, it sometimes miss the original line
+            };
+
+            const originalPosition = smc.originalPositionFor(position);
+            source.originalFilename = originalPosition.source && originalPosition.source.replace(/\\/g, '/');  // Ensure folder separator to be url-friendly
+            source.originalLine = originalPosition.line;
+            if (!originalPosition.source) {
+                if (line &&
+                    isValidLine) {
+
+                    ret.hits++;
+                }
+
+                return;
+            }
```

Comment:
This is the only real new logic in this PR.

It registers an automatic hit for any line that did not exist in a file, thus disregarding the actual coverage. The line can still have no hits when viewed in eg. the html reporter (but it's white since there is no 'miss'). I expect this is probably the least intrusive way to implement this. It would be nice to have a third "ignored" counter to represent these lines though, which could also be used when comments are used to ignore coverage.

The main problem with this approach, is that it artificially increases the coverage and an eg. `lab -t 95` run can pass or fail depending on how many lines the current version of typescript has added. Here it would be better to completely discount the lines from the coverage calculation.

Reply:
Changed it to reduce the number of lines instead of increase the number of hits.